modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/source/ExtensionSourceObjectFactory.java

```diff
@@ -109,6 +117,21 @@ public ExtensionMessageSource doGetObject() {
     });
   }
 
+  private Boolean calculatePrimaryNodeOnly(ResolverSet nonCallbackParameters) {
+    if (sourceModel.runsOnPrimaryNodeOnly()) {
+      return true;
+    }
+    try {
+      return (Boolean) nonCallbackParameters.getResolvers().get(PRIMARY_NODE_ONLY_PARAMETER_NAME)
```

Comment:
don't assume tha tthe resolver surely exists

Reply:
What would you say should happen if the resolver is not  there?

Reply:
return false;